### ISSUE TITLE
Incorrect calculation of the doc frequency cutoff threshold (STC) [CARROT-908]

### ISSUE THREAD

**Issue by dweiss** [carrot2/carrot2#1107]
In the class STCClusteringAlgorithm,there has a method called effectivePhraseLength which used to calculate "effective phrase length". And it has the following code:

```Java
if (tf < lower || tf > upper)
{
    continue;
}
```

tf is the total number of the word in documents,but i think is should be a number of documnts containing the word.Because upper = 0.9 \* documents.size(),if tf > upper,this shows that the word is common across all documents,then don't count it.

---
Issue: CARROT-908 (migrated from JIRA), created by Jianyun He, resolved May 14 2012

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Yes, indeed this is a bug – thanks for reporting! It should read (instead of tf):

```Java
                final int docCount = context.allWords.tfByDocument[termIndex].length / 2; 
                if (docCount < lower || docCount > upper)
                {
                    continue;
                }
```

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Fixed in trunk (3.6.0).
